### PR TITLE
#169257413 Comments on Travel requests

### PR DESCRIPTION
**Pull request by gildniy**
#### What does this PR do?
Enables users and managers to comment on travel requests
#### Description of Task to be completed?
 - [x] Created `comment` model and `comments` table
 - [x] Created relationship of table comments to table `users`, and table comments to table `requests`
 - [x] Create an end-point and controller to post comment on user Trip request
 - [x] Display comments on related travel request
###### To create a comment
- Clone the app on the local machine
- run `npm i` to install the dependencies
- Create a trip `/api/v1/trips/oneway`
- At that point copy the `id` in the returned data (which is the id of the request made)
- By going to `/api/v1/requests/:requestId/comment` in postman make a post with:
```
{
    "comment": "This is an example comment!"
}
```
###### To view comments on the request
- Go to the request endpoint in postman 

A list of all comments on the request will be available in the `comments` attribute on the returned request object.

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#169257413](https://www.pivotaltracker.com/story/show/169257413)
#### Screenshots (if appropriate)
- Trip / Request creation
image

- Trip comment creation
image

- Comments listing on trip request
1. 
image

2.
 image

#### Questions:
N/A

### CODE REVIEW

File: src/services/Comment.service.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+   * @param {string} requestId - email of the user
+   * @param {string} comment - email of the user
+   * @returns {object} user data from database
+   */
+  async postRequestComment(userId, requestId, comment) {
+    const requestData = await db.request.findOne({ where: { id: requestId } });
+
+
+    if (!requestData) {
+      throw new AppError('The request with the ID provided is not available.', 404);
```
Comment (timolinn) [inline src/services/Comment.service.js]: I think you can change error message to: `request resource not found`

File: src/services/Comment.service.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    const userLineManagerData = await db.user.findOne({
+      where: { id: request.userId },
+      attributes: ['lineManagerId']
+    });
+
+    const lineManagerId = userLineManagerData && userLineManagerData.dataValues.lineManagerId;
+
+    if (userId !== lineManagerId && userId !== request.userId) {
+      throw new AppError('Only the owner and his manager can comment on a request!', 403);
```
Comment (timolinn) [inline src/services/Comment.service.js]: I think this error message maybe better: `Unauthorised: you are not authorised to perform this operation`.

Comment (gildniy) [reply src/services/Comment.service.js]: > Unauthorised: you are not authorised to perform this operation

Fixed.

File: src/services/Comment.service.js (hunk 1 above)
Comment (gildniy) [reply src/services/Comment.service.js]: Fixed.

Comment (gildniy) [reply src/services/Comment.service.js]: > I think you can change error message to: `request resource not found`

Fixed.

**Review by timolinn — DISMISSED** [review]
Great work @gildniy

**Review by nshaphan — COMMENTED** [review]
Great work @gildniy, but I would suggest you make your endpoint look like `/requests/:requestId/comment` it's not a good practice to format endpoint  like`/:requestId/comment`

**Review by Mastel22 — DISMISSED** [review]
Tested and it's working. Great job

**Comment by gildniy** [comment]
> Great work @gildniy, but I would suggest you make your endpoint look like `/requests/:requestId/comment` it's not a good practice to format endpoint like`/:requestId/comment`

Thanks  @nshaphan for the suggestion, you're right as this format will inform about which type of comment will that be. Let me implement that.

**Review by Mastel22 — DISMISSED** [review]

**Review by nshaphan — DISMISSED** [review]
Great work @gildniy

**Review by timolinn — DISMISSED** [review]

**Comment by timolinn** [comment]
@gildniy Please fix conflicts.

**Review by timolinn — APPROVED** [review]

**Review by Mastel22 — APPROVED** [review]

**Review by nshaphan — APPROVED** [review]

**Pull request merged**